zio-http/src/test/scala/zio/http/TestServer.scala

```diff
@@ -0,0 +1,155 @@
+package zio.http
```

Comment:
Rather than putting this into `zio-http/src/test`, I think we need a new top-level project, e.g.: `zio-http-testkit`, so a user can rely on that (published) project in their own tests.